service/backend/pchain/backend.go

```diff
@@ -0,0 +1,111 @@
+package pchain
+
+import (
+	"context"
+
+	"github.com/ava-labs/avalanchego/codec"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/crypto"
+	"github.com/ava-labs/avalanchego/vms/platformvm/blocks"
+	"github.com/ava-labs/avalanchego/vms/platformvm/txs"
+	"github.com/coinbase/rosetta-sdk-go/types"
+
+	"github.com/ava-labs/avalanche-rosetta/client"
+	pmapper "github.com/ava-labs/avalanche-rosetta/mapper/pchain"
+	"github.com/ava-labs/avalanche-rosetta/service"
+	"github.com/ava-labs/avalanche-rosetta/service/backend/pchain/indexer"
+)
+
+var (
+	_ service.ConstructionBackend = &Backend{}
+	_ service.NetworkBackend      = &Backend{}
+	_ service.AccountBackend      = &Backend{}
+	_ service.BlockBackend        = &Backend{}
+)
+
+type Backend struct {
+	fac                    *crypto.FactorySECP256K1R
+	networkIdentifier      *types.NetworkIdentifier
+	pClient                client.PChainClient
+	indexerParser          indexer.Parser
+	getUTXOsPageSize       uint32
+	codec                  codec.Manager
+	codecVersion           uint16
+	genesisBlock           *indexer.ParsedGenesisBlock
+	genesisBlockIdentifier *types.BlockIdentifier
+	chainIDs               map[string]string
+	avaxAssetID            ids.ID
+}
+
+func NewBackend(
+	pClient client.PChainClient,
+	indexerParser indexer.Parser,
+	assetID ids.ID,
+	networkIdentifier *types.NetworkIdentifier,
+) *Backend {
+	return &Backend{
+		fac:               &crypto.FactorySECP256K1R{},
+		networkIdentifier: networkIdentifier,
+		pClient:           pClient,
+		getUTXOsPageSize:  1024,
+		codec:             blocks.Codec,
+		codecVersion:      txs.Version,
```

Comment:
TL;DR 
```
codec:        blocks.Codec,
codecVersion: blocks.Version,
```

nit: we have cleaned up platformvm to ease up codecs usage. Here you can now only use blocks.Codec/Version and forget about txs.Codec

Reply:
Awesome, will push an update with this today.